### PR TITLE
8285965: TestScenarios.java does not check for "<!-- safepoint while printing -->" correctly

### PR DESCRIPTION
**Pull request by chhagedorn**
This is another rare occurrence of `<!-- safepoint while printing -->` that is not handled correctly by `TestScenarios.java`. 

We wrongly search this safepoint message in the test VM output with `getTestVMOutput()`:

https://github.com/openjdk/jdk/blob/9c2548414c71b4caaad6ad9e1b122f474e705300/test/hotspot/jtreg/testlibrary_tests/ir_framework/tests/Utils.java#L44-L53

But this does not help since the IR matcher is parsing the `hotspot_pid` file for IR matching and not the test VM output. We could therefore find this safepoint message in the `hotspod_pid` file and bail out of IR matching while the test VM output does not contain it. This lets `TestScenarios.java` fail. 

The fix we did for other IR framework tests is to redirect the output of the JTreg test VM itself to a stream in order to search it for `<!-- safepoint while printing -->`. We are dumping this message as part of a warning when the IR matcher bails out:

https://github.com/openjdk/jdk/blob/9c2548414c71b4caaad6ad9e1b122f474e705300/test/hotspot/jtreg/compiler/lib/ir_framework/driver/irmatching/IRMatcher.java#L86-L96

Output for the reported failure:
```
Scenario #3 - [-XX:TLABRefillWasteFraction=53]:
[...]
Found <!-- safepoint while printing -->, bail out of IR matching
```

I suggest to use the same fix for `TestScenarios`.

Thanks,
Christian

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.java.net/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8285965](https://bugs.openjdk.org/browse/JDK-8285965): TestScenarios.java does not check for "" correctly

### Reviewers
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - **Reviewer**)
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8647/head:pull/8647` \
`$ git checkout pull/8647`

Update a local copy of the PR: \
`$ git checkout pull/8647` \
`$ git pull https://git.openjdk.java.net/jdk pull/8647/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8647`

View PR using the GUI difftool: \
`$ git pr show -t 8647`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8647.diff">https://git.openjdk.java.net/jdk/pull/8647.diff</a>

</details>

### CODE REVIEW

**Review by TobiHartmann — APPROVED** [review]
Looks good to me.

**Comment by chhagedorn** [comment]
Thanks Tobias for your review!

**Comment by chhagedorn** [comment]
Ping - may I get a second review for this?

**Review by vnkozlov — APPROVED** [review]
Good.

**Comment by chhagedorn** [comment]
Thanks Vladimir for your review!

**Pull request closed without merging**
